### ISSUE TITLE
Don't allow multiple concurrent sessions for same user account 

### ISSUE THREAD

**Issue by Knkkyl** [nextcloud/server#44170]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Multiple concurrent sessions should not be allowed from different devices for single account.
 

### Steps to reproduce

 Login to Nextcloud web interface with admin account from multiple devices simultaneously 

### Expected behavior

If session for admin account already exists, then either new session should not be allowed or kill the previous session.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by solracsf** [comment]
Could you elaborate why?

**Comment by Knkkyl** [comment]
Assuming an attacker gained access to an account, the simplest solution would be to change the password of said account. But what if the user could not log in to change the password because the session was already active somewhere else? The user is now locked out completely and the hacker has unlimited access.

The biggest risk is that session tokens can be stolen. In this case, multiple devices will use the same session, so to your backend, that's still just one concurrent login.

**Comment by nickvergessen** [comment]
When someone got compromised the way to go is to drop all auth tokens for that user.

This can be done via the command line:
```
sudo -u www-data php occ user:auth-tokens:list <user>
sudo -u www-data php occ user:auth-tokens:delete <user>
```

Or when logging in as the user navigating to `/index.php/settings/user/security` and revoking all the tokens:
![grafik](https://github.com/nextcloud/server/assets/213943/f89dfbef-0b22-4f17-b6c5-4b66e80694bd)
To revoke the "existing" session, just hit the logout button.

On that note, changing the password and email requires password confirmation, so a stolen session is not enough.

Additionally you should think about setting up 2FA. But binding a session to something else is just doomed to create problems along the way.

---

Having said all that, if you are scared on that level, you should **never** be running 27.1.2 at this point (but you seem to as per https://github.com/nextcloud/server/issues/44169 ). The current version is 27.1.7
You should also consider updating to 28.0.3

**Comment by solracsf** [comment]
@nickvergessen are those commands documented anywhere? 🤔 

```
# occ user:auth-tokens:list <username>

  There are no commands defined in the "user:auth-tokens" namespace.

  Did you mean this?
      user
```

**Comment by joshtrichards** [comment]
> are those commands documented anywhere? 🤔

Nope, let's fix that: nextcloud/documentation#12331
